pkg/syncer/cnsoperator/controller/csinodetopology/csinodetopology_controller.go

```diff
@@ -0,0 +1,336 @@
+	"sigs.k8s.io/vsphere-csi-driver/pkg/csi/service/logger"
+	csinodetopologyv1alpha1 "sigs.k8s.io/vsphere-csi-driver/pkg/internalapis/csinodetopology/v1alpha1"
+	k8s "sigs.k8s.io/vsphere-csi-driver/pkg/kubernetes"
+	"sigs.k8s.io/vsphere-csi-driver/pkg/syncer"
+)
+
+const defaultMaxWorkerThreadsForCSINodeTopology = 1
+
+// backOffDuration is a map of csinodetopology instance name to the time after which a request
+// for this instance will be requeued.
+// Initialized to 1 second for new instances and for instances whose latest reconcile
+// operation succeeded.
+// If the reconcile fails, backoff is incremented exponentially.
+var (
+	backOffDuration         map[string]time.Duration
+	backOffDurationMapMutex = sync.Mutex{}
+)
+
+// Add creates a new CSINodeTopology Controller and adds it to the Manager, ConfigurationInfo
+// and VirtualCenterTypes. The Manager will set fields on the Controller
+// and Start it when the Manager is Started.
+func Add(mgr manager.Manager, clusterFlavor cnstypes.CnsClusterFlavor,
+	configInfo *cnsconfig.ConfigurationInfo, volumeManager volumes.Manager) error {
+	ctx, log := logger.GetNewContextWithLogger()
+	if clusterFlavor != cnstypes.CnsClusterFlavorVanilla {
+		log.Debug("Not initializing the CSINodetopology Controller as it is not a Vanilla CSI deployment")
+		return nil
+	}
+
+	coCommonInterface, err := commonco.GetContainerOrchestratorInterface(ctx, common.Kubernetes, clusterFlavor, &syncer.COInitParams)
+	if err != nil {
+		log.Errorf("failed to create CO agnostic interface. Err: %v", err)
+		return err
+	}
+	if !coCommonInterface.IsFSSEnabled(ctx, common.ImprovedVolumeTopology) {
+		log.Infof("Not initializing the CSINodetopology Controller as %s FSS is disabled", common.ImprovedVolumeTopology)
+		return nil
+	}
+
+	// Initialize kubernetes client
+	k8sclient, err := k8s.NewClient(ctx)
+	if err != nil {
+		log.Errorf("Creating Kubernetes client failed. Err: %v", err)
+		return err
+	}
+	// eventBroadcaster broadcasts events on csinodetopology instances to the event sink
+	eventBroadcaster := record.NewBroadcaster()
+	eventBroadcaster.StartRecordingToSink(
+		&typedcorev1.EventSinkImpl{
+			Interface: k8sclient.CoreV1().Events(""),
+		},
+	)
+	recorder := eventBroadcaster.NewRecorder(scheme.Scheme, corev1.EventSource{Component: csinodetopologyv1alpha1.GroupName})
+	return add(mgr, newReconciler(mgr, clusterFlavor, configInfo, recorder))
+}
+
+// newReconciler returns a new reconcile.Reconciler
+func newReconciler(mgr manager.Manager, clusterFlavor cnstypes.CnsClusterFlavor,
+	configInfo *cnsconfig.ConfigurationInfo, recorder record.EventRecorder) reconcile.Reconciler {
+	return &ReconcileCSINodeTopology{client: mgr.GetClient(), scheme: mgr.GetScheme(),
+		clusterFlavor: clusterFlavor, configInfo: configInfo, recorder: recorder}
+}
+
+// add adds a new Controller to mgr with r as the reconcile.Reconciler
+func add(mgr manager.Manager, r reconcile.Reconciler) error {
+	_, log := logger.GetNewContextWithLogger()
```

Comment:
Since the context is ignored here, can you use `GetLoggerWithNoContext`

Reply:
Changed it